pkg/types/aws/validation/machinepool.go

```diff
@@ -43,3 +45,14 @@ func ValidateAMIID(platform *aws.Platform, p *aws.MachinePool, fldPath *field.Pa
 	}
 	return allErrs
 }
+
+// ValidateMachinePoolArchitecture checks that a valid architecture is set for a machine pool.
+func ValidateMachinePoolArchitecture(pool *types.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	index := sort.SearchStrings(types.ArchitectureListAWS, string(pool.Architecture))
```

Comment:
This is fragile because it requires `types.ArchitectureListAWS` to be sorted.
I recommend you set up a map of acceptable values like [here](https://github.com/openshift/installer/blob/4e60a8e317a164d3645bf370f303094fe2e18f8c/pkg/types/validation/machinepools.go#L43-L47).

Reply:
yup.done